File: src/modules/organization/validators/organization_delete_validator.py
from src.modules.organization.models import Organization
from src.modules.company.models import Company
from src.modules.user.models import User
from src.modules.role.models import Role
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.exceptions.conflict_exception import ConflictException


def organization_delete_validator(organization_id: int):

    organization = Organization.objects.filter(
        id=organization_id, deleted_at=None
    ).first()

    if organization is None:
        raise NotFoundException("Organização não encontrada!")

    has_companies = Company.objects.filter(
        organization__id=organization_id, deleted_at=None
    ).first()

    if has_companies is not None:
        raise ConflictException(
            "Não foi possível deletar organização por que possui empresas vinculadas!"
        )

    has_users = User.objects.filter(
        organization__id=organization_id, deleted_at=None
    ).first()

    if has_users is not None:
        raise ConflictException(
            "Não foi possível deletar organização por que possui usuários vinculados!"
        )

    has_roles = Role.objects.filter(
        organization__id=organization_id, deleted_at=None
    ).first()

    if has_roles is not None:
        raise ConflictException(
            "Não foi possível deletar organização por que possui perfis vinculados!"
        )

File: src/modules/organization/validators/organization-delete.validator.ts
import { ConflictException, NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';

export async function organizationDeleteValidator(organizationId: number) {
  const prisma = new PrismaService();

  const organization = await prisma.organization.findFirst({
    where: {
      id: organizationId,
      deletedAt: null,
    },
  });

  if (!organization) throw new NotFoundException('Organização não encontrada!');

  const hasCompanies = await prisma.company.findFirst({
    where: {
      organizationId: organizationId,
      deletedAt: null,
    },
  });

  if (hasCompanies)
    throw new ConflictException(
      'Não foi possível deletar organização por que possui empresas vinculadas!',
    );

  const hasUsers = await prisma.user.findFirst({
    where: {
      organizationId: organizationId,
      deletedAt: null,
    },
  });

  if (hasUsers)
    throw new ConflictException(
      'Não foi possível deletar organização por que possui usuários vinculados!',
    );

  const hasRoles = await prisma.role.findFirst({
    where: {
      organizationId: organizationId,
      deletedAt: null,
    },
  });

  if (hasRoles)
    throw new ConflictException(
      'Não foi possível deletar organização por que possui perfis vinculados!',
    );
}
